### PR TITLE
Optimize the use of beehive context

### PR DESCRIPTION
**Pull request by kadisi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

 /kind cleanup
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Optimize the use of beehive context
```

### CODE REVIEW

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go [hunk 1]
```diff
@@ -20,8 +20,12 @@ var (
 	once    sync.Once
 )
 
-// GetContext gets global context instance
-func GetContext(contextType string) *Context {
+func init()  {
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: Do we need this `init` here? We have already call the `InitContext` in `StartModule` directly.

File: staging/src/github.com/kubeedge/beehive/pkg/core/module.go [hunk 2]
```diff
@@ -64,6 +64,8 @@ func isModuleEnabled(m string) bool {
 type moduleChangeCallback struct{}
 
 func (cb moduleChangeCallback) Callback(k string, v interface{}) {
+	beehiveContext.InitContext(beehiveContext.MsgCtxTypeChannel)
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/module.go]: no need `InitContext` here?

Comment (kadisi) [reply staging/src/github.com/kubeedge/beehive/pkg/core/module.go]: ok

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go (hunk 1 above)
Comment (kadisi) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: ok

**Pull request merged**